```
./src/ocean/text/convert/Formatter.d(468): Error: template `ocean.text.convert.Formatter_test.__unittest_L580_C1.SysTime.toString` cannot deduce function from argument types `!()(void delegate(const(char[]) e) @system)`, candidates are:
./src/ocean/text/convert/Formatter_test.d(585):        `toString(W)(ref W writer)`
./src/ocean/text/convert/Formatter.d(279): Error: template instance `ocean.text.convert.Formatter.handle!(SysTime)` error instantiating
./src/ocean/text/convert/Formatter.d(137):        instantiated from here: `sformat!(SysTime)`
./src/ocean/text/convert/Formatter_test.d(587):        instantiated from here: `format!(SysTime)`
make: *** [/home/runner/work/ocean/ocean/build/prod/tmp/allunittests] Error 1
submodules/makd/Makd.mak:574: recipe for target '/home/runner/work/ocean/ocean/build/prod/tmp/allunittests' failed
```

Bingo

--- a/src/ocean/text/convert/Formatter.d
+++ b/src/ocean/text/convert/Formatter.d
@@ -464,9 +464,13 @@ private void handle (T) (T v, FormatInfo f, scope FormatterSink sf, scope ElemSi
     // toString hook: Give priority to the non-allocating one
     // Note: sink `toString` overload should take a `scope` delegate
     else static if (is(typeof(v.toString(sf))))
+    {
+        // Some `toString` signatures expect a `ref` sink
+        scope sinkRef = (in cstring e) { se(e, f); };
         nullWrapper(&v,
-                    v.toString((in cstring e) { se(e, f); }),
+                    v.toString(sinkRef),
                     se("null", f));
+    }
     else static if (is(typeof(v.toString()) : cstring))
         nullWrapper(&v, se(v.toString(), f), se("null", f));
     else static if (is(T == interface))

--- a/src/ocean/text/convert/Formatter_test.d
+++ b/src/ocean/text/convert/Formatter_test.d
@@ -576,3 +576,13 @@ unittest
     }
     myFunc("Hello World");
 }
+
+unittest
+{
+    // Some types (e.g. Phobos's `SysTime` take a sink by ref)
+    static struct SysTime
+    {
+        void toString (W) (ref W writer) { writer("Hello World"); }
+    }
+    assert(format("{}", SysTime.init) == "Hello World");
+}

